keps/sig-node/3759-sidecar-containers/README.md

```diff
@@ -0,0 +1,1239 @@
+in back-to-back releases.
+
+**For non-optional features moving to GA, the graduation criteria must include
+[conformance tests].**
+
+[conformance tests]: https://git.k8s.io/community/contributors/devel/sig-architecture/conformance-tests.md
+
+#### Deprecation
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality that deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+### Upgrade / Downgrade Strategy
+
+#### Upgrade strategy
+
+Existing sidecars will still work as intended, as in the past we didn't
+recognize this type of container.
+
+#### Downgrade strategy
+
+Due to the new field added to initContainers to turn them into sidecars,
+downgrading to the version without this feature will not be possible -  the Pod
+spec will be rejected by the control plane and all kubelets during
+unmarshalling. Workloads will have to be deleted and recreated with the old way
+of handling sidecars.
+
+The same problem will arise if the user wants to deploy such workloads on older
+clusters, or clusters without the feature flag enabled.
+
+
+### Version Skew Strategy
+
+Version skew is possible between the control plane and worker nodes as both
+should be aware of the new field used to flag sidecars inside initContainers.
+
+Therefore all cluster nodes, including master nodes, must be upgraded before the
+user can deploy sidecars using the new syntax.
+
+Also, since the feature flag applies to both kubelet and the control plane,
+similarly all cluster nodes need to have it enabled before deploying Pods with
+sidecars.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+Feature gate (also fill in values in kep.yaml)
+- Feature gate name: `SidecarContainers`
+- Components depending on the feature gate: kubelet, kube-apiserver
```

Comment:
I think this belongs under  “**How can this feature be enabled / disabled in a live cluster?**”

Reply:
fixed